src/Compilers/CSharp/Portable/Symbols/Source/SourceParameterSymbolBase.cs

```diff
@@ -60,7 +58,7 @@ public sealed override Symbol ContainingSymbol
             get { return _containingSymbol; }
         }
 
-        public sealed override AssemblySymbol ContainingAssembly
+        public sealed override AssemblySymbol? ContainingAssembly
```

Comment:
Can this actually be null for a source parameter?

Reply:
It is very hard to guarantee that right now, in large part because our symbol tree has large annotation gaps. I think it's likely true that this is non-null. I'm not sure what would / wouldn't give me the confidence to change the annotation. 

One path I did consider is flipping it to annotated as non-null **and** add a `Debug.Assert` in the single constructor here to guarantee it. If that passed all unit tests would you be satisfied that we'd done enough dilligence to flip the annotation here? I don't mind doing that experiment.

Reply:
Yes, I think that would satisfy me.

Reply:
Okay let's see how this goes ...

Reply:
Unit tests all passed so we'll go with the non-null annotation here.